tonic_validate/classes/run.py

```diff
@@ -5,7 +5,7 @@
 
 @dataclass
 class RunData:
-    scores: dict[str, float]
+    scores: dict[str, float | None]
```

Comment:
This change is giving me an error.
<img width="970" alt="Screenshot 2024-02-15 at 3 30 05 PM" src="https://github.com/TonicAI/tonic_validate/assets/90645649/7f53f2cc-dc06-4614-ac42-303c1c55ddb9">

Reply:
I'm going to take this opportunity to replace the `dict` typing with `Dict` typing and `|` typing with `Union` to see if it helps.

Reply:
Changing the typing fixed it.

Reply:
Yeah, might be due to different python versions as that's valid in some newer versions of python